durabletask/worker.py

```diff
@@ -353,44 +353,92 @@ def is_replaying(self) -> bool:
+                      input: Optional[TInput] = None,
+                      retry_policy: Optional[task.RetryPolicy] = None) -> task.Task[TOutput]:
         id = self.next_sequence_number()
-        name = activity if isinstance(activity, str) else task.get_name(activity)
-        encoded_input = shared.to_json(input) if input else None
-        action = ph.new_schedule_task_action(id, name, encoded_input)
-        self._pending_actions[id] = action
-
-        activity_task = task.CompletableTask[TOutput]()
-        self._pending_tasks[id] = activity_task
-        return activity_task
+        
+        self.call_activity_util(id, activity, input, retry_policy)
+        return self._pending_tasks.get(id, task.CompletableTask())
+
+    def call_activity_util(self, id: Optional[int],
+                           activity: Union[task.Activity[TInput, TOutput], str],
+                           input: Optional[TInput] = None,
+                           retry_policy: Optional[task.RetryPolicy] = None,
+                           act_task: Optional[task.CompletableTask[TOutput]] = None):
+            if id is None:
+                id = self.next_sequence_number()
+            name = activity if isinstance(activity, str) else task.get_name(activity)
+                    
+            if act_task is None:
+                encoded_input = shared.to_json(input) if input else None
+            else:
+                encoded_input = input.__str__()
+            action = ph.new_schedule_task_action(id, name, encoded_input)
+            self._pending_actions[id] = action
+            if act_task is None:
+                if retry_policy is None:
+                    activity_task = task.CompletableTask[TOutput]()
+                    self._pending_tasks[id] = activity_task
+                else:
+                    activity_task = task.RetryableTask[TOutput](retry_policy=retry_policy, action=action)
+                    self._pending_tasks[id] = activity_task
+            else:
+                self._pending_tasks[id] = act_task
 
     def call_sub_orchestrator(self, orchestrator: task.Orchestrator[TInput, TOutput], *,
-                              input: Union[TInput, None] = None,
-                              instance_id: Union[str, None] = None) -> task.Task[TOutput]:
+                              input: Optional[TInput] = None,
+                              instance_id: Optional[str] = None,
+                              retry_policy: Optional[task.RetryPolicy] = None) -> task.Task[TOutput]:
         id = self.next_sequence_number()
-        name = task.get_name(orchestrator)
+        orchestrator_name = task.get_name(orchestrator)
+        self.call_sub_orchestrator_util(id, orchestrator_name, input=input, instance_id=instance_id,
+                                        retry_policy=retry_policy)
+        return self._pending_tasks.get(id, task.CompletableTask())
+    
+    def call_sub_orchestrator_util(self, id: Optional[int],
```

Comment:
Prefer `call_sub_orchestrator_helper`.

Reply:
Merged both helper methods and calling call_activity_function_helper.